include/zephyr/rtio/rtio.h

```diff
@@ -774,6 +921,61 @@ static inline void rtio_cqe_submit(struct rtio *r, int result, void *userdata)
 #endif
 }
 
+static inline struct rtio_sqe *rtio_sqe_handle_mempool(struct rtio *r, struct rtio_sqe *sqe)
+{
+#ifndef CONFIG_RTIO_SYS_MEM_BLOCKS
+	UNUSED_ARG(r);
+#else
+	__ASSERT_NO_MSG(r != NULL);
+	__ASSERT_NO_MSG(sqe != NULL);
+
+	if (sqe->op != RTIO_OP_RX || sqe->buf != NULL || r->mempool == NULL) {
+		return sqe;
+	}
+
+//	if (sys_mem_blocks_alloc(r->mempool, 1, (void**)&sqe->buf)) {
+//		/* Failed to allocate, return sqe and let the consumer decide what to do */
+//		return sqe;
+//	}
+//
+//	sqe->buf_len = r->mempool_blk_size;
+	sqe->flags |= RTIO_SQE_MEMPOOL_BUFFER;
```

Comment:
This should be set in prep_memblock_read, the next two member sets shouldn't be needed if you have the iodev_sqe in the device (which has a reference to this context)